include/interface/blas1_interface_sycl.hpp

```diff
@@ -94,222 +101,250 @@ void _copy(Executor<ExecutorType> ex, int _N, vector_view<T, ContainerT> _vx,
+  auto assignOp1 =
+      make_addAssignReduction(my_rs, prdOp, localSize, localSize * nWG);
+  ex.reduce(assignOp1);
 #ifdef VERBOSE
-  my_vx.printH("VX");
-  my_vy.printH("VY");
+  my_rs.printH("VR");
 #endif  //  VERBOSE
 }
 
 /**
- * \brief SCAL scales a vector by a constant
+ * \briefCompute the inner product of two vectors with extended
+    precision accumulation and result.
  *
  * @param Executor<ExecutorType> ex
  * @param _vx  VectorView
  * @param _incx Increment in X axis
+ * @param _vx  VectorView
+ * @param _incy Increment in Y axis
  */
-template <typename ExecutorType, typename T, typename ContainerT>
-void _scal(Executor<ExecutorType> ex, int _N, T _alpha,
-           vector_view<T, ContainerT> _vx, int _incx) {
-  auto my_vx = vector_view<T, ContainerT>(_vx, _vx.getDisp(), _incx, _N);
+template <typename ExecutorType, typename T>
+T _dot(Executor<ExecutorType> &ex, int _N, T *_vx, int _incx, T *_vy,
+       int _incy) {
 #ifdef VERBOSE
-  std::cout << "alpha = " << _alpha << std::endl;
   my_vx.printH("VX");
+  my_vy.printH("VY");
 #endif  //  VERBOSE
-  auto scalOp = make_op<ScalarOp, prdOp2_struct>(_alpha, my_vx);
-  auto assignOp = make_op<Assign>(my_vx, scalOp);
-  ex.execute(assignOp);
+  auto val_ptr = ex.template allocate<T>(1);
+  auto res = std::vector<T>(1);
+  _dot(ex, _N, _vx, _incx, _vy, _incy, val_ptr);
+  ex.copy_to_host(val_ptr, res.data(), 1);
+  ex.template deallocate<T>(val_ptr);
+
 #ifdef VERBOSE
-  my_vx.printH("VX");
+  std::cout << "val = " << res << std::endl;
 #endif  //  VERBOSE
+  return res[0];
 }
 
 /**
- * \briefCompute the inner product of two vectors with extended
-    precision accumulation and result.
- *
- * @param Executor<ExecutorType> ex
+ * \brief IAMAX finds the index of the first element having maximum
  * @param _vx  VectorView
  * @param _incx Increment in X axis
- * @param _vx  VectorView
- * @param _incy Increment in Y axis
  */
-template <typename ExecutorType, typename T, typename ContainerT>
-T _dot(Executor<ExecutorType> ex, int _N, vector_view<T, ContainerT> _vx,
-       int _incx, vector_view<T, ContainerT> _vy, int _incy) {
-  auto my_vx = vector_view<T, ContainerT>(_vx, _vx.getDisp(), _incx, _N);
-  auto my_vy = vector_view<T, ContainerT>(_vy, _vy.getDisp(), _incy, _N);
+template <typename ExecutorType, typename T, typename I>
+void _iamax(Executor<ExecutorType> &ex, int _N, T *_vx, int _incx, I *_rs) {
+  using RHS =
+      vector_view<T, typename Executor<ExecutorType>::template ContainerT<T>>;
+  using RHS1 =
+      vector_view<I, typename Executor<ExecutorType>::template ContainerT<I>>;
+  auto vx_container = ex.get_buffer(_vx);
+  RHS my_vx(vx_container, ex.get_offset(_vx), _incx, _N);
+  auto rs_container = ex.get_buffer(_rs);
+  RHS1 my_rs(rs_container, ex.get_offset(_rs), 1, 1);
 #ifdef VERBOSE
   my_vx.printH("VX");
-  my_vy.printH("VY");
 #endif  //  VERBOSE
-  auto prdOp = make_op<BinaryOp, prdOp2_struct>(my_vx, my_vy);
+  size_t localSize = 256, nWG = 512;
```

Comment:
is this set in stone? mark it const and explain how they can be set

Reply:
create a github issue to make this device-selectable

Reply:
I have put  TODO to fix it on the code. I have also created the issue on the github as well.